File: MonEndoVue.Server/Models/Medicament.cs
using TypeGen.Core.TypeAnnotations;

namespace MonEndoVue.Server.Models;

[ExportTsInterface]
public class Medicament
{
    public int Id { get; set; }
    public int CarnetSanteId { get; set; }
    public string Nom { get; set; }
    public string Posologie { get; set; }
    public bool TraitementEnCours { get; set; }
    [TsOptional]
    public DateTime DateDebutTraitement { get; set; }
    [TsOptional]
    public DateTime? DateFinTraitement { get; set; }
}

File: monendovue.client/src/features/medicament/types/medicament.ts
import type { TypeTraitement } from "./type-traitement";

export interface Medicament {
    id: number;
    carnetSanteId: number;
    nom: string;
    type: TypeTraitement;
    posologie?: string;
    traitementEnCours: boolean;
    dateDebutTraitement?: Date;
    dateFinTraitement?: Date;
}
